### DISCUSSION TITLE
HasReactionSubstructMatch not as expected

### DISCUSSION THREAD

**Discussion by shirte** [rdkit/rdkit#6396] [Q&A] [answered]
I would like to divide chemical reactions into groups and I am using the function ```HasReactionSubstractMatch``` for that purpose: 

```python
from rdkit.Chem import AllChem, rdChemReactions

oxidation = AllChem.ReactionFromSmarts('[*:1]>>[*:1]=O')
rxn = AllChem.ReactionFromSmarts('[*:1]CC(=O)C>>[*:1]CC(=O)C')
result = rdChemReactions.HasReactionSubstructMatch(rxn, oxidation)
print(result) # True
```

According to the code, the first reaction (top) is a template of the second reaction (bottom):

![image](https://github.com/rdkit/rdkit/assets/1706443/e7e174a2-70bd-4567-a155-3f7a41509535)
![image](https://github.com/rdkit/rdkit/assets/1706443/bbce7b65-62a7-4226-9f5c-3686cd6ae053)

However, I would expect ```HasReactionSubstructMatch``` to return "False", because the second reaction does actually nothing. Is this the intended behaviour of ```HasReactionSubstructMatch```? If yes, what would be an alternative approach to check whether one reaction is a specific variant of another reaction? 

My rdkit version is 2023.03.1 (python 3.11.3).

**Comment by greglandrum** [comment, marked as answer]
`HasReactionSubstructMatch()` checks to be sure that each reactant and product has a substructure match. That is the case here.

The RDKit doesn't have anything builtin to determine whether or not one reaction is a more specific variant of another. In general that would be equivalent to determining whether or not a SMARTS pattern is a more specific variant of another, and that's a very tricky problem.
